FixedHammers/FixedHammers.lua

```diff
@@ -0,0 +1,62 @@
+ModUtil.RegisterMod("FixedHammers")
+
+local config = {
+  ModName = "Fixed Hammers",
+  UseFixedHammers = true
+}
+
+if ModConfigMenu then
+  ModConfigMenu.Register(config)
+end
+
+local function GetEligibleHammers()
+
+  local loot = DeepCopyTable( LootData["WeaponUpgrade"] )
+  loot.RarityChances = {}
+  loot.ForceCommon = true
+  return GetEligibleUpgrades({}, loot, LootData["WeaponUpgrade"])
+end
+
+ModUtil.WrapBaseFunction("StartNewRun", function(baseFunc, ...)
+  -- initialize everything else first
+  local run = baseFunc(...)
+
+  -- reset the mod
+  FixedHammers.Hammers = {}
+
+  -- determine eligible hammers (for the current weapon)
+  local eligibleHammers = GetEligibleHammers()
+
+  -- shuffle the hammers, creating two permutations
+  FixedHammers.Hammers[1] = FYShuffle(eligibleHammers)
+  FixedHammers.Hammers[2] = FYShuffle(eligibleHammers)
+
+  return run
+end, FixedHammers)
+
+ModUtil.WrapBaseFunction("SetTraitsOnLoot", function(baseFunc, lootData, args)
+  if lootData.Name == "WeaponUpgrade" and config.UseFixedHammers and FixedHammers.Hammers then
+    local previousHammers = CurrentRun.LootTypeHistory.WeaponUpgrade or 0
+    local hammersInOrder = FixedHammers.Hammers[previousHammers + 1]
+
+    local eligibleHammers = GetEligibleHammers()
+    local upgradeOptions = {}
+    local upgradesSelected = 0
```

Comment:
Not sure if it matters much, but maybe we should align on a consistent code style.  I'm also comfortable trying to treat this as more of a modpack where each sub module should just be consistent within itself, but either way we should make a conscious decision.  Museus and I were vaguely using snake case for locals